### ISSUE TITLE
ros_visualizer example throws errors when using IPv6 and OS dependent behavior

### ISSUE THREAD

**Issue by JohannesProgrammiert** [jhelovuo/RustDDS#205]
Great project, I consider using this at my company.
However, at one of my PCs I only have IPv6 available. When I run the `ros_visualizer` example with

```
cargo run --example ros_visualizer
```

About once a second I get

```
WARN rustdds::network::udp_sender - send_to_locator - send_to [<ipv6>]:<port> : 0s {code: 97, kind Uncategorized, message: "Address family not supported by protocol"}
```

On Manjaro (where the behavior from above was observed), the `shapes_demo` example works, also with the RTI Connext demo but not with OpenDDS. The OpenDDS demo does not seem to interact with any other interop example other than itself.

On Debian, rustDDS (and openDDS) do not work at all. There is simply no discovery and only HEARTBEAT messages are sent. Any ideas?

Another question, does this library support shared memory for IPC?

**Comment by jhelovuo** [comment]
The suggestion of disabling IPv6 for now is a good one. Can you make on PR on that?

Did you check all the basic things on Debian, such as firewalling? Does `ping` work to multicast addresses, e.g. 224.0.0.1 ?

**Comment by flxo** [comment]
Hello,

are there any updates regarding IPv6 on the roadmap? IPv6 is marked unimplemented [here](https://github.com/jhelovuo/RustDDS/blob/08e9edbeb36e4435958b982d3f6b28b0b1b1ee82/src/network/udp_sender.rs#L77) and [here](https://github.com/jhelovuo/RustDDS/blob/08e9edbeb36e4435958b982d3f6b28b0b1b1ee82/src/network/udp_listener.rs#L134).

Thanks!

**Comment by jhelovuo** [comment]
> are there any updates regarding IPv6 on the roadmap? IPv6 is marked unimplemented [here](https://github.com/jhelovuo/RustDDS/blob/08e9edbeb36e4435958b982d3f6b28b0b1b1ee82/src/network/udp_sender.rs#L77) and [here](https://github.com/jhelovuo/RustDDS/blob/08e9edbeb36e4435958b982d3f6b28b0b1b1ee82/src/network/udp_listener.rs#L134).

Not really. It is still not on our internal roadmap. If there were some commercial interest, then that could change, but currently no-one else seems to be interested.

Pull Requests are still welcome.
